src/modules/dropdown.js

```diff
@@ -381,9 +384,12 @@ layui.define(['jquery', 'laytpl', 'lay', 'util'], function(exports){
     
     // 若存在已打开的面板元素，则移除
     if(elemPrev){
-      elemPrev.data('prevElem') && (
-        elemPrev.data('prevElem').data(MOD_INDEX +'_opened', false)
-      );
+      var prevElem = elemPrev.data('prevElem');
+      if(prevElem){
+        prevElem.data(MOD_INDEX +'_opened', false);
+        var prevOpt = prevElem.data(MOD_INDEX +'_on_close');
+        typeof prevOpt.close === 'function' && prevOpt.close(prevElem);
+      }
```

Comment:
这里可能是被 `prevElem` 误导了，直接操作原型链是否也能达到目的呢，晨星可做进一步验证。
```suggestion
      var prevElem = elemPrev.data('prevElem');
      if (prevElem) {
        prevElem.data(MOD_INDEX +'_opened', false);
      }
```